ddtrace/span.py

```diff
@@ -105,8 +105,7 @@ def __init__(
         that it was created in. Using a ``Span`` from within a child process
         could result in a deadlock or unexpected behavior.
 
-        :param ddtrace.Tracer tracer: the tracer that will submit this span when
-            finished.
+        :param ddtrace.Tracer tracer: deprecated. Pass ``None`` instead.
```

Comment:
In principle there is no need to force passing `None` if this is just a deprecation that we are gracefully handling. I'd rather document that the value will be ignored and a warning is emitted in case of the value being not `None`. WDYT?

Reply:
this is a good though, we could update `__init__` to just not set `_tracer` right?

e.g.

```python
def __init__(self, tracer, name, ...):
    if tracer is not None:
        # log a deprecation warning that the argument is now ignored
    # do not set `self.tracer` or `self._tracer`, just let it stay as the default `None`, even if they pass in a tracer to `__init__` ?
```


@P403n1x87 what do you think of this suggestion?

Reply:
👍  makes sense to me

Reply:
We could raise a deprecation warning when the tracer argument is set. I think we should always set the `self._tracer` attribute otherwise Span.tracer would raise an exception (Although I could use `getattr(self, '_tracer', None)` in `def tracer(self):` if that's preferable).

```
def __init__(self, tracer, name, ...):
        self._tracer = None
        if tracer is not None:
            deprecate(
                message="tracer parameter is deprecated. Use Span(tracer=None, ...) to initialize a Span.",
                removal_version="1.0.0",
            )
            self._tracer = tracer  # type: Tracer
```

Reply:
hmm... I was going to suggest saying that the argument is now ignored and always set `self._tracer = None`, but that breaks the API if someone is expecting to be able to do `Span(tracer, ...)` and then `span.tracer` to get that tracer back...


From your example, we should keep `self._tracer` as an `Optional[Tracer]` and just `return self._tracer` in the property instead of the `getattr`.